src/Microsoft.IdentityModel.Protocol.Extensions/OpenIdConnectMessage.cs

```diff
@@ -96,7 +96,7 @@ public OpenIdConnectMessage(IEnumerable<KeyValuePair<string, string[]>> paramete
 
             foreach (KeyValuePair<string, string[]> keyValue in parameters)
             {
-                if (keyValue.Value != null)
+                if (keyValue.Value != null && !string.IsNullOrWhiteSpace(keyValue.Key))
```

Comment:
Do we need this check here? Few lines below this line of code, we check for the null value. Does it have any perf benefit checking it here? If so, we should remove the check from the other line.